cmd/govim/code_lens.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/govim/govim"
+	"github.com/govim/govim/cmd/govim/internal/golang_org_x_tools/lsp/protocol"
+	"github.com/govim/govim/cmd/govim/internal/golang_org_x_tools/lsp/source"
+)
+
+func (v *vimstate) toggleGCDetails(flags govim.CommandFlags, args ...string) error {
+	cb, _, err := v.bufCursorPos()
+	if err != nil {
+		return fmt.Errorf("failed to determine cursor position: %v", err)
+	}
+	res, err := v.server.CodeLens(context.Background(), &protocol.CodeLensParams{
+		TextDocument: cb.ToTextDocumentIdentifier(),
+	})
+	if err != nil {
+		return fmt.Errorf("codeLens failed: %v", err)
+	}
+
+	for _, cl := range res {
```

Comment:
Can we get back multiple `CommanToggleDetails` commands? If so, please can you add a comment that explains what that means, given the current behaviour is a toggle?

Otherwise, we should assert that we get back at most one.

Reply:
Yeah, good point. Just another case of me running best effort but that would hide potential bugs in gopls. Fixed.